### ISSUE TITLE
Add `_saveCookies` and `jar`

### ISSUE THREAD

**Issue by jsdevwork** [WhoopInc/supertest-as-promised#38]
Can we add `_saveCookies` and `jar` to the agent? This will do the trick. Would be happy to submit a PR.

```(javascript)
  out.agent = function () {
    var agent = supertest.agent.apply(null, arguments);
    var _agent = wrap(agent);
    _agent._saveCookies = agent._saveCookies;
    _agent.jar = agent.jar;
    return _agent;
  };
```

**Comment by benesch** [comment]
Good catch! Fixed another way, since it turns out we should just inherit from the agent prototype chain to expose all such properties, rather than special-casing the two you listed. Thanks for the report!

**Comment by benesch** [comment]
Released as v4.0.1. Cheers!
